core/selection.cpp

```diff
@@ -231,3 +237,19 @@ extern "C" void select_newest_visible_dive()
 	// No visible dive -> deselect all
 	select_single_dive(nullptr);
 }
+
+extern "C" void select_trip(struct dive_trip *trip)
+{
+	if (trip && !trip->selected) {
+		trip->selected = true;
+		amount_trips_selected++;
```

Comment:
in our UI logic, selecting a trip always implies selecting all the dives in the trip.
Would it make sense to do this hear, or would this confuse other layers and we really just worry about the helper data structure for trips?
As I write the question, I think I know the answer... :-)

Reply:
Yes, that was the original idea. However, I'm not sure about the correct way to implement this, so first I wanted to start with the "easy" version. The problem is that we get a loop:

UI selection change -> dive list -> Core selects more items -> dive list

We end up with a selection change in the divelist, while a selection change is active. What we could do is
1) a "apply_selection_change()" function that returns a list of additional items that the dive list has to select.
2) simply intercepting all UI selection changes, sending them to the core and let the core rebuild the selection. I.e. never apply selections directly.

Reply:
I can see the advantage of the much simpler approach to not deal with causing new selections